src/UCTSearch.cpp

```diff
@@ -195,6 +195,47 @@ void UCTSearch::dump_stats(FastState & state, UCTNode & parent) {
             node->get_score() * 100.0f,
             pv.c_str());
     }
+    tree_stats(parent);
+}
+
+void tree_stats_helper(UCTNode* node, size_t depth,
+                       size_t& nodes, size_t& non_leaf_nodes,
+                       size_t& depth_sum, size_t& max_depth,
+                       size_t& children_count) {
+    nodes += 1;
+    non_leaf_nodes += node->get_visits() > 1;
+    depth_sum += depth;
+    if (depth > max_depth) max_depth = depth;
+
+    for (const auto& child : node->get_children()) {
+        if (!child->first_visit()) children_count += 1;
+
+        tree_stats_helper(child.get(), depth+1,
+                          nodes, non_leaf_nodes, depth_sum,
+                          max_depth, children_count);
+    }
+}
+
+void UCTSearch::tree_stats(UCTNode& node) {
+    if (cfg_quiet || !node.has_children()) {
+        return;
+    }
+
+    size_t nodes = 0;
+    size_t non_leaf_nodes = 0;
+    size_t depth_sum = 0;
+    size_t max_depth = 0;
+    size_t children_count = 0;
+    tree_stats_helper(&node, 0,
+                      nodes, non_leaf_nodes, depth_sum,
+                      max_depth, children_count);
+
+    if (nodes > 0) {
+        myprintf(" %.1f average depth, %d max depth\n",
```

Comment:
I like one space here to line them up with the output from dump_stats. I'm also open to removing it to line up with the "X visits, Y nodes" printed from `think()`